src/datahike/index/persistent_set.cljc

```diff
@@ -199,21 +199,24 @@
       (uuid (mapv (comp vec seq) (.keys node))))
     (uuid)))
 
-(defrecord CachedStorage [store config cache stats pending-writes]
+(defrecord CachedStorage [store config cache stats pending-writes cost-center-fn]
```

Comment:
The function can track a compute budget, e.g. for an API user and can throw when the budget is used up. I did not expose an interface to it yet to see whether it will prove useful. In our blockchainesque experiments with the datopia project we were able to render the access to the database permissionless by validating incoming data with Datalog (as a smart contract language) and this is complementary functionality to also make sure that such contracts get interrupted as soon as their budget is consumed. But I think this might be of interest also as an independent functionality just to meter an API.

Reply:
Thanks for the explanation :-)